### DISCUSSION TITLE
[WEB SDK] Playing all Tracks from Album using Player - Howto requested

### DISCUSSION THREAD

**Discussion by Ro-Lem** [tidal-music/discussions#117] [Q&A]
Hi,
could you please provide an example, how to play all tracks from an album using `@tidal-music/player`?

I have a proof-of-concept working for one track - but it stops playing after the first track of the album.
Only the first MediaProduct is playing - then playback stops.

Could you please push me to the right direction and give me a hint, what is going wrong here?

Beside that, I only found PLAY and PAUSE - I am missing STOP, NEXT, PREV - here I would also be very happy go get an hint how to archive this.

Thank you in advance.

```
import * as auth from "@tidal-music/auth";
import * as player from "@tidal-music/player";

....

async function playTidal() {

  const clientId = "CLIENT-ID";
  const clientSecret = "CLIENT-SECRET";

    await auth.init({
      clientId,
      clientSecret,
      credentialsStorageKey: 'key',
      scopes: [],
    });

    player.setCredentialsProvider(auth.credentialsProvider);
      
// MediaProducts have the following structure
    // productId, productType, sourceId, sourceType 
      await player.load(mediaProducts[0]);
      await player.setNext(mediaProducts[1]);
      await player.setNext(mediaProducts[2]);
      await player.setNext(mediaProducts[3]);

      await player.play();
// Plays, but stops after first Track / MediaProduct
}
```

**Comment by stefanmoro** [comment]
Hey @Ro-Lem ,

The SDK does not currently include queuing logic (which would implement next/prev), so that is something that you would have to implemented in your own app. Thanks.

**Reply by Ro-Lem** [reply]
Hi stefanmoro,
thank you for your answer.

I already tried to implement a queue by using the `player.setNext()`-function as shown in my post.

Unfortunately without success - the player only plays the first title and stops. It seems to ignore the next ones.  
I would be very grateful for a hint where the error lies in my approach.

Thank you in advance.
